### ISSUE TITLE
adafruit_bno055.py throws exception (error 19) on register write during initialization

### ISSUE THREAD

**Issue by hukuzatuna** [adafruit/Adafruit_CircuitPython_BNO055#2]
Environment: Metro M0 Express, CircuitPython 2.0 rc2 "latest" build as of last night's release.
Output:
Traceback (most recent call last):
  File "code.py", line 71, in <module>
  File "/lib/adafruit_bno055.py", line 111, in __init__
  File "/lib/adafruit_bno055.py", line 176, in init
  File "/lib/adafruit_bno055.py", line 117, in _write_register
  File "/lib/adafruit_bno055.py", line 117, in _write_register
  File "libraries/helpers/bus_device/adafruit_bus_device/i2c_device.py", line 94, in write
OSError: 19
 
(If this actually belongs in the bundled library issues, close it and let me know and I'll open an issue there. Thanks! -Phil)

**Comment by tannewt** [comment]
Is this still an issue?

**Comment by hukuzatuna** [comment]
I'm not sure. I'm at work and don't have my device. I will check this evening!

**Comment by tannewt** [comment]
I just tested on CircuitPython close to 2.2 and it worked fine.

**Comment by caternuson** [comment]
Seems to be back:
```python
Adafruit CircuitPython 2.2.4 on 2018-03-07; Adafruit Feather M0 Express with samd21g18
>>> import board
>>> import busio
>>> import adafruit_bno055
>>> i2c = busio.I2C(board.SCL, board.SDA)
>>> sensor = adafruit_bno055.BNO055(i2c)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "adafruit_bno055.py", line 127, in __init__
  File "adafruit_bno055.py", line 138, in _write_register
  File "adafruit_bno055.py", line 138, in _write_register
  File "adafruit_bus_device/i2c_device.py", line 102, in write
OSError: 19
>>>
```

**Comment by caternuson** [comment]
This seems to be related to how the wait-for-reset is being done in the driver code. From that BNO055 datasheet, there's a POR time, from reset to config mode, of 650 ms typ. (thanks to jepler in discord for pointing this out). This isn't being respected in the code which checks for chip id as a way to determine if reset is complete. For some reason, that check returns way before 650 ms and the subsequent i2c comms fail and cause the above traceback.

**Comment by tannewt** [comment]
Great find! Care to make a PR?

**Comment by caternuson** [comment]
I will if @deshipu doesn't want to. Probably various ways it could be fixed. He might be more familiar with quirks and trade offs with this sensor.

**Comment by caternuson** [comment]
Hopefully fixed by #10
